src/shared/components/mutationTable/MutationTable.tsx

```diff
@@ -0,0 +1,340 @@
+import {ICosmicData} from "shared/model/Cosmic";
+import AnnotationColumnFormatter from "./column/AnnotationColumnFormatter";
+import {IMyCancerGenomeData} from "shared/model/MyCancerGenome";
+import {IHotspotData} from "shared/model/CancerHotspots";
+import {IOncoKbData} from "shared/model/OncoKB";
+import {IMutSigData} from "shared/model/MutSig";
+import DiscreteCNACache from "shared/cache/DiscreteCNACache";
+import OncoKbEvidenceCache from "shared/cache/OncoKbEvidenceCache";
+import PmidCache from "shared/cache/PmidCache";
+
+export interface IMutationTableProps {
+    sampleIds?:string[];
+    discreteCNACache?:DiscreteCNACache;
+    oncoKbEvidenceCache?:OncoKbEvidenceCache;
+    pmidCache?:PmidCache
+    mutSigData?:IMutSigData;
+    myCancerGenomeData?: IMyCancerGenomeData;
+    hotspots?: IHotspotData;
+    cosmicData?:ICosmicData;
+    oncoKbData?:IOncoKbData;
+    mrnaExprRankGeneticProfileId?:string;
+    discreteCNAGeneticProfileId?:string;
+    columns:MutationTableColumnType[];
+    data:Mutation[][];
+}
+
+export enum MutationTableColumnType {
+    SAMPLE_ID,
+    TUMORS,
+    GENE,
+    PROTEIN_CHANGE,
+    CHROMOSOME,
+    START_POS,
+    END_POS,
+    REF_ALLELE,
+    VAR_ALLELE,
+    MUTATION_STATUS,
+    VALIDATION_STATUS,
+    MUTATION_TYPE,
+    CENTER,
+    TUMOR_ALLELE_FREQ,
+    MUTATION_ASSESSOR,
+    ANNOTATION,
+    COSMIC,
+    COPY_NUM,
+    MRNA_EXPR,
+    COHORT,
+    REF_READS_N,
+    VAR_READS_N,
+    REF_READS,
+    VAR_READS
+}
+
+type MutationTableColumn = Column<Mutation[]>&{order?:number};
+
+export class MutationTableComponent extends MSKTable<Mutation[]> {
+}
+
+export function getSpanForDataField(data:Mutation[], dataField:string) {
+    let contents = getTextForDataField(data, dataField);
+    return (<span>{contents}</span>);
+}
+
+export function getTextForDataField(data:Mutation[], dataField:string) {
+    let text = "";
+    if (data.length > 0 && data[0].hasOwnProperty(dataField)) {
+        text = (data[0] as any)[dataField];
+    }
+    return text;
+}
+
+export function defaultFilter(data:Mutation[], dataField:string, filterStringUpper:string):boolean {
+    if (data.length > 0) {
+        return data.reduce((match:boolean, next:Mutation)=>{
+            const val = (next as any)[dataField];
+            if (val) {
+                return match || ((val.toUpperCase().indexOf(filterStringUpper) > -1));
+            } else {
+                return match;
+            }
+        }, false);
+    } else {
+        return false;
+    }
+}
+
+@observer
+export default class MutationTable<P extends IMutationTableProps> extends React.Component<P, {}> {
+    @observable protected _columns:{[columnEnum:number]:MutationTableColumn};
+
+    constructor(props:P)
+    {
+        super(props);
+        this._columns = {};
+        this.generateColumns();
+    }
+
+    protected getSamples():string[] {
+        if (this.props.sampleIds) {
+            return this.props.sampleIds as string[];
```

Comment:
I think you don't need that because if PVTableProps has all the properties of TableProps, then PVTableProps can be passed in as TableProps even if it has other things too. Not 100% sure on that though.